src/main/java/gregtech/api/recipes/ingredients/IntCircuitIngredient.java

```diff
@@ -13,6 +13,19 @@ public class IntCircuitIngredient extends GTRecipeItemInput {
     public static final int CIRCUIT_MAX = 32;
     private final int matchingConfigurations;
 
+    public static IntCircuitIngredient getOrCreate(IntCircuitIngredient ri, int i) {
+        return (IntCircuitIngredient) getFromCache(new IntCircuitIngredient(getCircuitConfiguration(ri.getInputStacks()[0])));
```

Comment:
Why do you have to call `getCircuitConfiguration()`, couldn't you just do `ri.matchingConfigurations`?